vhdbuilder/packer/post-install-dependencies.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+OS=$(sort -r /etc/*-release | gawk 'match($0, /^(ID_LIKE=(coreos)|ID=(.*))$/, a) { print toupper(a[2] a[3]); exit }')
+UBUNTU_OS_NAME="UBUNTU"
+
+source /home/packer/provision_installs.sh
+source /home/packer/provision_installs_distro.sh
+source /home/packer/provision_source.sh
+source /home/packer/provision_source_distro.sh
+source /home/packer/tool_installs.sh
+source /home/packer/tool_installs_distro.sh
+
+CPU_ARCH=$(getCPUArch)  #amd64 or arm64
+VHD_LOGS_FILEPATH=/opt/azure/vhd-install.complete
+
+if [[ $OS == $MARINER_OS_NAME ]]; then
+  if grep -q "kata" <<< "$FEATURE_FLAGS"; then
+    enableMarinerKata
+  else
+    # Leave automatic package update disabled for the kata image
+    enableDNFAutomatic
+  fi
+fi
+
+if [[ $OS == $UBUNTU_OS_NAME ]]; then
+  # shellcheck disable=SC2021
+  current_kernel="$(uname -r | cut -d- -f-2)"
+  dpkg --get-selections | grep -e "linux-\(headers\|modules\|image\)" | grep -v "$current_kernel" | tr -s '[[:space:]]' | tr '\t' ' ' | cut -d' ' -f1 | xargs -I{} apt-get remove -yq {}
+
+  # strip old kernels/packages
+  apt-get -y autoclean || exit 1
+  apt-get -y autoremove --purge || exit 1
+  apt-get -y clean || exit 1
+  # remove apport
+  apt-get purge --auto-remove apport open-vm-tools -y
+fi
+
+# shellcheck disable=SC2129
+echo "kubelet/kubectl downloaded:" >> ${VHD_LOGS_FILEPATH}
+ls -ltr /usr/local/bin/* >> ${VHD_LOGS_FILEPATH}
+
+# shellcheck disable=SC2010
+ls -ltr /dev/* | grep sgx >>  ${VHD_LOGS_FILEPATH} 
+
+echo -e "=== Installed Packages Begin\n$(listInstalledPackages)\n=== Installed Packages End" >> ${VHD_LOGS_FILEPATH}
+
+echo "Disk usage:" >> ${VHD_LOGS_FILEPATH}
+df -h >> ${VHD_LOGS_FILEPATH}
+
+
+# warn at 75% space taken
+[ -s $(df -P | grep '/dev/sda1' | awk '0+$5 >= 75 {print}') ] || echo "WARNING: 75% of /dev/sda1 is used" >> ${VHD_LOGS_FILEPATH}
```

Comment:
unrelated: can you create a backlog task to do two things here:
- use disk detection by label for OS disk (/dev/sda not reliable!)
- check capacity of OS disk by label
- hardcode max capacity as 30 GB. I want to change os disk size to 512GB to alleviate IO latency during VHD builds. So need to avoid relying on the disk size itself, only current util.

Reply:
not blocking for merge

Reply:
new items created